fastly/block_fastly_service_v1_logging_cloudfiles_test.go

```diff
@@ -191,6 +231,15 @@ func testAccCheckFastlyServiceV1CloudfilesAttributes(service *gofastly.ServiceDe
 					// these ahead of time
 					el.CreatedAt = nil
 					el.UpdatedAt = nil
+
+					// Ignore VCL attributes for Compute and set to whatever is returned from the API.
+					if serviceType == ServiceTypeCompute {
+						el.FormatVersion = e.FormatVersion
+						el.Format = e.Format
+						el.ResponseCondition = e.ResponseCondition
+						el.Placement = e.Placement
+					}
```

Comment:
Maybe we could have `testService` that could centralize this logic.

I'm thinking something with a signature like

```
func testService(t *testing.T, tf func(svc interface{}, serviceType string)) {
}
```
Where in this case, the lines [here](https://github.com/terraform-providers/terraform-provider-fastly/pull/285/files#diff-93f92f31d2de63f8b27ff734af2935f3R194)

Would look like:

```golang
testService(testAccCheckFastlyServiceV1CloudfilesAttributes(..., ServiceTypeWasm))
```

(Again, **not a blocker**, just a thought).

Reply:
As discussed, we're due to rethink testing generally so we should give this a go